hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/logical/MapScanLogicalRule.java

```diff
@@ -37,6 +39,11 @@ private MapScanLogicalRule() {
     @Override
     public RelNode convert(RelNode rel) {
         LogicalTableScan scan = (LogicalTableScan) rel;
+        HazelcastTable table = scan.getTable().unwrap(HazelcastTable.class);
+        if (!(table.getTarget() instanceof AbstractMapTable)) {
```

Comment:
This check should be performed in the rule pattern, rather than the rule's body. 
An example of how to add a custom predicate to rule patterns could be found in the `ProjectIntoScanLogicalRule` constructor.

Reply:
fixed